skorch/hf.py

```diff
@@ -0,0 +1,800 @@
+"""Classes to work with huggingface libraries
+
+E.g. transformers or tokenizers
+
+This module should be treated as a leaf node in the dependency tree, i.e. no
+other skorch modules should depend on these classes or import from here. Even
+so, don't import any Huggingface libraries on the root level because skorch
+should not depend on them.
+
+"""
+
+import os
+from copy import deepcopy
+from operator import itemgetter
+
+import numpy as np
+from sklearn.base import BaseEstimator, TransformerMixin
+
+from skorch.utils import check_is_fitted, params_for
+
+
+class _HuggingfaceTokenizerBase(BaseEstimator, TransformerMixin):
+    """Base class for yet to train and pretrained tokenizers
+
+    Implements the ``vocabulary_`` attribute and the methods
+    ``get_feature_names``, ``transform``, and ``inverse_transform``.
+
+    Subclasses should implement the ``fit``  method.
+
+    """
+    @property
+    def vocabulary_(self):
+        check_is_fitted(self, ['fast_tokenizer_'])
```

Comment:
Interesting. I would expect a property to raise `AttrubuteError` rather than `NotFittedError` if the estimator hasn't been fit. For instance, `logreg.coef_` raises an `AttrubuteError` when you haven't fit the model.

You could either leave it out, or use `@available_if` if you want (although I'm not sure if the latter works with a property).

Reply:
Yes, you're right, `AttributeError` is more in line with expectations and with other sklearn transformers. I'll change the check accordingly.